docs/src/tutorials/manifold.md

```diff
@@ -0,0 +1,198 @@
+Actually these two lines are the default implementation for [`exp`](@ref exp(M::Manifold, p, X)).
+Note that for a unified interface, the manifold `M` is _always_ the first parameter, and the variable the result will be stored to in the mutating variants is _always_ the second parameter.
+
+Long story short: if possible, implement the mutating version [`exp!`](@ref exp!(M::Manifold, q, p, X)), you get the [`exp`](@ref exp(M::Manifold, p, X)) for free.
+Many functions that build upon basic functions employ the mutating variant, too, to avoid reallocations.
+
+## [Startup](@id manifold-tutorial-startup)
+
+As a start, let's load `ManifoldsBase.jl` and import the functions we consider throughout this tutorial.
+For implementing a manifold, loading the interface should suffice for quite some time.
+
+```@example manifold-tutorial
+using ManifoldsBase, LinearAlgebra, Test
+import ManifoldsBase: check_manifold_point, check_tangent_vector, manifold_dimension, exp!
+```
+
+## [Goal](@id manifold-tutorial-task)
+
+As an example, let's implement the sphere, but with a radius $r$. Since this radius is a property inherent to the manifold.
+The second information, we want to store is the dimension of the sphere, for example whether it's the 1-sphere, i.e. the circle, represented by vectors $p\in\mathbb R^2$ or the 2-sphere in $\mathbb R^3$.
+Since the latter might be something we want to [dispatch](https://en.wikipedia.org/wiki/Multiple_dispatch) on, we model it as a parameter of the type. Hence we define
+
+```@example manifold-tutorial
+"""
+    MySphere{N} <: Manifold{ℝ}
+
+Define an `n`-sphere of radius `r`. Construct by `MySphere(radius,n)
+"""
+struct MySphere{N} <: Manifold{ManifoldsBase.ℝ} where {N}
+    radius::Float64
+end
+MySphere(radius, n) = MySphere{n}(radius)
+Base.show(io::IO, M::MySphere{n}) where {n} = print(io, "MySphere($(M.radius),$n)")
+nothing #hide
+```
+
+Here, the last line just provides a nicer print of a variable of that type
+The manifold should only contain information that it might require independently of its points or tangents.
+Now we can already initialize our Manifold, we use later, the $2$-sphere of radius $1.5$.
+
+```@example manifold-tutorial
+S = MySphere(1.5,2)
+```
+
+## [Checking points and tangents](@id manifold-tutorial-checks)
+
+If we have now a point, represented as an array, we would first like to check, that it is a valid point on the manifold.
+For this one can use the easy interface [`is_manifold_point`](@ref is_manifold_point(M::Manifold, p; kwargs...)). This internally uses [`check_manifold_point`](@ref check_manifold_point(M, p; kwargs...)).
+This is what we want to implement.
+We have to return the error, if `p` is not on `M` and `nothing` otherwise.
+
+We have to check two things: that a point `p` is a vector with `N+1` entries and it's norm is the desired radius.
+To spare a few lines, we can use [lazy evaluation](https://en.wikipedia.org/wiki/Lazy_evaluation) instead of `if` statements.
```

Comment:
```suggestion
To spare a few lines, we can use [short-circuit evaluation](https://docs.julialang.org/en/v1/manual/control-flow/#Short-Circuit-Evaluation-1) instead of `if` statements.
```